apps/src/lib/levelbuilder/course-editor/CourseEditor.js

```diff
@@ -263,17 +274,40 @@ export default class CourseEditor extends Component {
         </CollapsibleEditorSection>
 
         <CollapsibleEditorSection title="Teacher Resources">
-          <div>
-            Select the Teacher Resources buttons you'd like to have show up on
-            the top of the course overview page
-          </div>
-          <ResourcesEditor
-            inputStyle={styles.input}
-            resources={teacherResources}
-            updateTeacherResources={teacherResources =>
-              this.setState({teacherResources})
-            }
-          />
+          {this.props.migratedResources && (
+            <input
+              type="hidden"
+              name="resourceIds[]"
+              value={this.props.migratedResources.map(r => r.id)}
+            />
+          )}
+          {this.props.useMigratedResources ? (
```

Comment:
If you did want to reduce the number of times you check useMigratedResources throughout the code, one idea would be that once you have set resources and migratedResources, to just check the contents of those props when deciding what to display, and trust that whatever code higher up the stack did not accidentally set both